src/CommunityToolkit/Xamarin.CommunityToolkit/Views/SideMenuView/SideMenuView.shared.cs

```diff
@@ -332,13 +356,17 @@ void PerformUpdate(bool isAnimated = true)
 				SetOverlayViewInputTransparent(state);
 				return;
 			}
+
+			stateChangingEventManager.RaiseEvent(this, new SideMenuStateChangingEventArgs(State), nameof(StateChanging));
```

Comment:
events won't be called if isAnimated = false
Also, events will be called if the state wasn't changed (if the user drags menu a little and release the touch)

Reply:
I think if `isAnimated = false` we should just trigger the `StateChanged` event. That will follow the intent of this PR where `StateChanged` is triggered after the animation or state has changed. I'll have updated this part in my next code push.

Your 2nd point about events firing even though the state hasn't changed is going to require a little more rules discussion. I just ran through it and it fires multiple events, yet they all return the same state. Maybe we can update the `State` Property to see if the new state is different than the previous state prior to applying it. 

Do you think it is best to simplify this API and just have it be `StateChanged`? Will trying to add `StateChanging` cause too many edge cases that will be difficult to maintain? In the gesture scenario you brought up there is no way for us to identify if the state is actually going to change, we can only make the assumption that the sidemenu is moving. Maybe we rename `StateChanging` to `MenuSliding` to indicate that the menu has started sliding but it hasn't completed. Another option is we could have a position delta event that is thrown every update cycle with a state. I have seen manipulation APIs like that in both WPF and UWP.

Reply:
I think this is covered in the comment below - let me know if there is anything else we need to do on this comment